### DISCUSSION TITLE
Building GrandOrgue from sources on macOS

### DISCUSSION THREAD

**Discussion by willeke1234** [GrandOrgue/grandorgue#1158] [Developers] [answered]
I'm trying to build GrandOrgue from sources on my Mac. I followed the instructions in BUILD.md, but I get errors:

```
grandorgue/src/grandorgue/sound/ports/GOSoundRtPort.cpp:121:12: error: 
      unknown type name 'RtAudioError'; did you mean 'RtAudioErrorType'?
          } catch (RtAudioError &e) {
                   ^~~~~~~~~~~~

grandorgue/src/grandorgue/sound/ports/GOSoundRtPort.cpp:291:29: error: 
      no member named 'probed' in 'RtAudio::DeviceInfo'
              if (!dev_info.probed)
                   ~~~~~~~~ ^
```

I can't find a definition of `RtAudioError` in the sources. Am I missing something?

**Comment by oleg68** [comment, marked as answer]
How did you get GrandOrgue sourcecode? If you used ``git clone --recurse-submodules https://github.com/GrandOrgue/grandorgue.git``. then `RtAudioError` is defined in `submodules/RtAudio/RtAudio.h`

**Comment by willeke1234** [comment]
I used Source Control in Xcode and the Manually section in BUILD.md. The link downloads RtAudio 6.0.0beta1. I did `git clone`, the errors are gone but I got
```
CMake Error at /Volumes/Macintosh HD/Users/willeke/Documents/Projects/GrandOrgue/GrandOrgue dev/grandorgue/cmake/FixupBundle.cmake:34 (include):
  include could not find requested file:

    /Volumes/Macintosh\ HD/Users/willeke/Documents/Projects/GrandOrgue/GrandOrgue\ dev/gobuild/cmake.status
```

cmake.status exists but the path contains spaces and I think the `\` shouldn't be there. I'd rather not remove the space from "Macintosh HD". If I remove some quotes in CopyDependencies.cmake and change

`COMMAND ${CMAKE_COMMAND} -Dstatusfile="${statusfile}" -Dtarget="${targetfile}" -Dsearchdirs="${cmd_searchdirs}" -P "${CMAKE_SOURCE_DIR}/cmake/FixupBundle.cmake"`

to

`COMMAND ${CMAKE_COMMAND} -Dstatusfile=${statusfile} -Dtarget=${targetfile} -Dsearchdirs="${cmd_searchdirs}" -P "${CMAKE_SOURCE_DIR}/cmake/FixupBundle.cmake"`

then GrandOrgue builds but I'm just guessing and I don't really know what I'm doing.
